spec.html

```diff
@@ -11059,6 +11060,26 @@ <h1>Examples of Automatic Semicolon Insertion</h1>
       <p>In the circumstance that an assignment statement must begin with a left parenthesis, it is a good idea for the programmer to provide an explicit semicolon at the end of the preceding statement rather than to rely on automatic semicolon insertion.</p>
     </emu-clause>
   </emu-clause>
+  <emu-clause id="sec-hazards-of-automatic-semicolon-insertion">
+    <h1>Hazards of Automatic Semicolon Insertion</h1>
+    <em>This section is non-normative.</em>
+    <p>ECMAScript programs can be written in a style with very few semicolons by relying on automatic semicolon insertion. However, as described above, semicolons are not inserted at every newline, and some of these cases which do not have an automatically inserted semicolon can be counter-intuitive or confusing.</p>
+
+    <p>As new syntactic features are added to ECMAScript, additional cases requiring explicit semicolons emerge over time. As such, consistently explicit semicolon use is recommended.</p>
```

Comment:
Just adding my $0.02 ... perhaps something more along the lines of:

```
Additional cases requiring explicit semicolons will emerge as new syntactic features are added to
ECMAScript. As such, automatic semicolon insertion should be used only when it is known to be
safe.
```

It gets the same basic point across but without the explicit recommendation of explicit over ASI.

Reply:
@jasnell I like the less aggressive wording, but it does imply that ASI usage will stay up to date both in tooling ([which historically has been a problem](https://github.com/rwaldron/tc39-notes/blob/master/es8/2017-11/nov-30.md#10ivf-class-fields-asi-discussion-and-resolution)) and that all ASI hazards be learned for all new syntaxes that are increasingly being in the awkward situation of creating such hazards. I think this removes the attempt of the PR to state a way for code to be written in a way that future (additive) changes do not change how the code works.

Reply:
A recommendation of explicit use of semicolons is just for consistency. Would be super weird to have semicolons mixed in with no semicolons.

Reply:
The explicit recommendation is the primary point of this PR.

Reply:
rephrased a bit